store/src/java/com/zimbra/cs/index/event/Event.java

```diff
@@ -0,0 +1,53 @@
+package com.zimbra.cs.index.event;
+
+import java.util.Map;
+
+public class Event {
+    private EventType eventType;
+    private long timestamp;
+    private Map<EventContextField, Object> context;
+
+    public enum EventType {
+        SENT, RECEIVED, READ, SEEN
+    }
+
+    public enum EventContextField {
+        USER_IDENTIFIER, SENDER, RECEIVER
```

Comment:
I think that USER_IDENTIFIER actually _can_ be common to all events and doesn't need to go here. In other words, the Event constructor would take an event type, accountId, timestamp, and context map.